File: javascript/razor.js
const prompt = require("prompt-sync")({ sigint: true });

const calculateTotalPayment = (purchaseAmount) => {
  let discount, discountAmount, totalPayment;

  if (purchaseAmount >= 200000 && purchaseAmount <= 1000000) {
    discount = 0.03; // 3% discount
  } else if (purchaseAmount > 1000000) {
    discount = 0.07; // 7% discount
  } else {
    discount = 0.0;
  }

  discountAmount = purchaseAmount * discount;
  totalPayment = purchaseAmount - discountAmount;
  return { discountAmount, totalPayment };
};

function processCustomers() {
  for (let choice = "Y"; choice === "Y" || choice === "y"; ) {
    let purchaseAmount = prompt(`Enter the purchase amount: `);
    purchaseAmount = parseFloat(purchaseAmount);

    const { discountAmount, totalPayment } =
      calculateTotalPayment(purchaseAmount);

    console.log(`Discounted Amount: $${discountAmount.toFixed(2)}`);
    console.log(`Total Payment after discount: $${totalPayment.toFixed(2)}\n`);

    choice = prompt("Do you want to continue (Y/N)? ");
  }
}

processCustomers();

File: src/razor.cpp
#include <iostream>

using namespace std;

int main()
{
    double discountAmount, purchaseAmount, totalPayment, discount;
    char choice;

    do
    {
        cout << "Enter Purchase Amount: ";
        cin >> purchaseAmount;

        if (purchaseAmount >= 200000 && purchaseAmount <= 1000000)
        {
            discount = 0.03;
        }
        else if (purchaseAmount > 1000000)
        {
            discount = 0.07;
        }
        else
        {
            discount = 0.00;
        }

        discountAmount = discount * purchaseAmount;
        totalPayment = purchaseAmount - discountAmount;

        cout << "Discounted Amount: " << discountAmount << endl;
        cout << "Total Payment after discount: " << totalPayment << endl;

        cout << "Calculate for another customer? (Y/N): ";
        cin >> choice;
    } while (choice == 'Y' || choice == 'y');

    return 0;
}
